modules/router-store/src/router_store_module.ts

```diff
@@ -279,7 +279,10 @@ export class StoreRouterConnectingModule {
 
     if (this.router.url !== this.storeState[this.stateKey].state.url) {
       this.navigationTriggeredByDispatch = true;
-      this.router.navigateByUrl(this.storeState[this.stateKey].state.url);
```

Comment:
This should be a separate PR. I don't think this will work for SSR apps also because you can't guarantee this will be finished before the next navigation starts.